app/views/investments/show.html.erb

```diff
@@ -20,15 +20,22 @@
       </div>
     <% end %>
 
-    <% if @investment.category == 'share' && @info["regularMarketChangePercent"].present? %>
+    <% if @investment.category  && @info && @info["regularMarketChangePercent"].present? %>
       <div>
-        <p><strong>Market change:</strong> <%= @info["regularMarketChangePercent"].round(2)%>%</p>
+        <% if @info["regularMarketChangePercent"].positive?%>
```

Comment:
I think it might be better to do the if statement around the style tag only so we don't have so much repeated code. What do you think?

Reply:
Oh, I didn't even think of that!

Reply:
I've tried to change it like that but it just gives syntax errors, I might have to look at it more.. I kinda wanted to do it on JS with an 'onload' eventlistener tbh